x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyout.tsx

```diff
@@ -60,17 +54,26 @@ export function AddSettingsFlyout({
       ? selectedConfig.settings.transaction_sample_rate.toString()
       : ''
   );
-  const { data: serviceNames = [], status: serviceNamesStatus } = useFetcher<
-    string[]
-  >(async () => (await loadAgentConfigurationServices()).sort(), [], {
-    preservePreviousResponse: false
-  });
-  const { data: environments = [], status: environmentStatus } = useFetcher<
-    Array<{ name: string; available: boolean }>
-  >(
+  const { data: serviceNames = [], status: serviceNamesStatus } = useFetcher(
+    async () =>
+      (await callApmApi({
+        pathname: '/api/apm/settings/agent-configuration/services'
+      })).sort(),
```

Comment:
I know this is not a new change, but can we move the `sort` to the backend?

Reply:
definitely!

Reply:
fixed